src/components/MainPage.js

```diff
@@ -51,60 +52,99 @@ const styles = () => ({
 };
 
 type State = {
   inputValue: number | string,
-  favorites: Array<number>,
+  activePage: number,
+  favorites: string,
 };
 
 class MainPage extends Component<Props, State> {
   componentDidMount() {
-    this.props.getMovies.movies();
+    const urlParamsValues = queryString.parse(this.props.location.search);
+    this.setState({ activePage: urlParamsValues.page || 1 });
+    this.props.getMovies.movies(
+      null,
+      urlParamsValues.page || 1,
+      'popularity.desc'
+    );
+    const favorites = localStorage.getItem('favorites') || JSON.stringify([]);
+    localStorage.setItem('favorites', favorites);
   }
 
   handleSearchChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
-    this.setState({ inputValue: event.target.value });
+    this.setState({
+      inputValue: event.target.value,
+      activePage: 1,
+    });
   };
 
   handleSubmit = (event: SyntheticInputEvent<HTMLInputElement>) => {
     event.preventDefault();
-    this.props.getMovies.movies(this.state.inputValue);
+    const urlParamsValues = queryString.parse(this.props.location.search);
+    this.setState({ activePage: urlParamsValues.page });
+    this.props.getMovies.movies(
+      this.state.inputValue,
+      urlParamsValues.page || this.state.activePage,
+      'popularity.desc'
+    );
   };
 
-  //TODO: add functionality to handle list of favorites movies and bottom navigation
-  handleBottomNavigationChange = (
-    event: SyntheticInputEvent<HTMLInputElement>
-  ) => {
-    event.preventDefault();
-    console.log('handleBottomNavigationChange');
+  handlePageChange = (pageNumber: number) => {
+    const { inputValue } = this.state;
+    const value = inputValue || null;
+    this.setState({ activePage: pageNumber });
+    this.props.getMovies.movies(value, pageNumber);
+    const history = createBrowserHistory({ basename: '/' });
+    history.push(`?page=${pageNumber}`);
   };
 
-  openListOfFavorites = (event: SyntheticInputEvent<HTMLInputElement>) => {
-    event.preventDefault();
-    console.log('openListOfFavorites');
+  addMovieToListOfFavorites = (id: number) => {
+    const favorites = localStorage.getItem('favorites');
+    if (favorites) {
+      const parsedFavorites = JSON.parse(favorites);
+      const duplicate = _.find(e => e === id, parsedFavorites);
+      if (duplicate !== id) {
+        parsedFavorites.push(id);
+        const favToString = JSON.stringify(parsedFavorites);
+        localStorage.setItem('favorites', favToString);
+      }
+    }
   };
 
-  addMovieToListOfFavorites = () => {
-    console.log('addMovieToListOfFavorites');
+  deleteMovieFromFavorites = (id: number) => {
+    let favorites = localStorage.getItem('favorites');
+    if (favorites) {
+      const parsedFavorites = JSON.parse(favorites);
+      const updatedList = _.filter(e => e !== id, parsedFavorites);
+      const updatedListToString = JSON.stringify(updatedList);
+      localStorage.setItem('favorites', updatedListToString);

```

Comment:
Try to use `_.flow` here - composition. You will avoid creation of redundant variables

Reply:
👀